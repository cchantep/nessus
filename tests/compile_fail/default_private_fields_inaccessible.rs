// This test verifies that private fields are not accessible by default (include_private = false)

use tnuctipun::{FieldWitnesses, MongoComparable};
use serde::{Deserialize, Serialize};
// Test struct with mixed visibility and default behavior (include_private = false)
#[derive(Debug, Clone, Serialize, Deserialize, FieldWitnesses, MongoComparable)]
struct DefaultBehaviorTest {
    pub name: String,      // public field - should be included
    email: String,         // private field - should be skipped
    pub age: i32,          // public field - should be included
    internal_id: u64,      // private field - should be skipped
}

fn main() {
    // These should fail to compile because private fields are skipped by default:
    let _ = defaultbehaviortest_fields::Email::field_name();
    let _ = defaultbehaviortest_fields::InternalId::field_name();
}
