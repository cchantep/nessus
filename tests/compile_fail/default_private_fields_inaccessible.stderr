error[E0433]: failed to resolve: could not find `Email` in `defaultbehaviortest_fields`
  --> tests/compile_fail/default_private_fields_inaccessible.rs:17:41
   |
17 |     let _ = defaultbehaviortest_fields::Email::field_name();
   |                                         ^^^^^ could not find `Email` in `defaultbehaviortest_fields`

error[E0433]: failed to resolve: could not find `InternalId` in `defaultbehaviortest_fields`
  --> tests/compile_fail/default_private_fields_inaccessible.rs:18:41
   |
18 |     let _ = defaultbehaviortest_fields::InternalId::field_name();
   |                                         ^^^^^^^^^^ could not find `InternalId` in `defaultbehaviortest_fields`

warning: unused import: `FieldName`
 --> tests/compile_fail/default_private_fields_inaccessible.rs:3:14
  |
3 | use nessus::{FieldName, FieldWitnesses, MongoComparable};
  |              ^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
